File: go-demo/services/heavy.go
package services

type Heavy struct{}

func NewHeavy() *Heavy {
	return &Heavy{}
}

func (factorial *Heavy) CPUBurn(n int) int {
	result := 1
	for i := 0; i < n; i++ {
		result *= i
	}
	return result
}

func (factorial *Heavy) MemoryBurn(n int) int {
	largeSlice := make([]uint32, n)
	for i := 0; i < n; i++ {
		largeSlice[i] = uint32(i)
	}
	var sum uint32 = 0
	for _, num := range largeSlice {
		sum += num
	}
	return int(sum)
}

File: dotnet-demo/DotnetDemo.Api/Heavy.cs
using System;

namespace DotnetDemo.Api;

public class Heavy
{
    public int CPUBurn(int n)
    {
        int result = 1;
        for(int i = 0; i < n; i++)
            result *= i;
        return result;
    }
    public long MemoryBurn(long n)
    {
        List<uint> largeList = new();
        for(int i = 0; i < n; i++)
            largeList.Add(Convert.ToUInt32(i));
        // return largeList.Sum();
        long sum = 0;
        foreach(var num in largeList)
            sum += num;
        return sum;
    }
}
